src/hyperion/external_interaction/callbacks/ispyb_callback_base.py

```diff
@@ -90,60 +103,130 @@ def activity_gated_event(self, doc: Event) -> Event:
-            )
-            if doc["data"]["dcm_energy_in_kev"]:
-                energy_ev = doc["data"]["dcm_energy_in_kev"] * 1000
-                self._event_driven_data_collection_info.wavelength = (
-                    convert_eV_to_angstrom(energy_ev)
+                ISPYB_LOGGER.info("Updating ispyb entry.")
+                self.ispyb_ids = self.update_deposition(
+                    self.params,
+                    scan_data_infos,
+                    self._sample_barcode,
                 )
-                # TODO 1173 Remove this once nexus_utils no longer needs wavelength_angstroms
-                self.params.hyperion_params.ispyb_params.current_energy_ev = energy_ev
+                ISPYB_LOGGER.info(f"Recieved ISPYB IDs: {self.ispyb_ids}")
+        return self._tag_doc(doc)
+
+    def _handle_ispyb_hardware_read(self, doc):
+        assert self._hwscan_data_collection_info
+        ISPYB_LOGGER.info("ISPyB handler received event from read hardware")
+        self._hwscan_data_collection_info.undulator_gap1 = doc["data"][
+            "undulator_current_gap"
+        ]
+        self._hwscan_data_collection_info.synchrotron_mode = doc["data"][
+            "synchrotron-synchrotron_mode"
+        ].value
+        self._hwscan_data_collection_info.slitgap_horizontal = doc["data"][
+            "s4_slit_gaps_xgap"
+        ]
+        self._hwscan_data_collection_info.slitgap_vertical = doc["data"][
+            "s4_slit_gaps_ygap"
+        ]
+        self._sample_barcode = doc["data"]["robot-barcode"]
+
+    def _handle_oav_snapshot_triggered(self, doc):
+        assert self.ispyb_ids.data_collection_ids, "No current data collection"
+        assert self.params, "ISPyB handler didn't recieve parameters!"
+        data = doc["data"]
+        data_collection_id = None
+        data_collection_info = DataCollectionInfo()
+        data_collection_info.xtal_snapshot1 = data.get(
+            "oav_snapshot_last_path_full_overlay"
+        )
+        data_collection_info.xtal_snapshot2 = data.get("oav_snapshot_last_path_outer")
+        data_collection_info.xtal_snapshot3 = data.get("oav_snapshot_last_saved_path")
+        data_collection_info.n_images = (
+            data["oav_snapshot_num_boxes_x"] * data["oav_snapshot_num_boxes_y"]
+        )
+        grid_scan_info = GridScanInfo(
+            upper_left_px=numpy.array(
+                [data["oav_snapshot_top_left_x"], data["oav_snapshot_top_left_y"]]
+            ),
+            x_steps=data["oav_snapshot_num_boxes_x"],
+            y_steps=data["oav_snapshot_num_boxes_y"],
+            # convert pixels back to mm
+            x_step_size_mm=data["oav_snapshot_box_width"]
+            * self.params.hyperion_params.ispyb_params.microns_per_pixel_x
+            / 1000,
+            y_step_size_mm=data["oav_snapshot_box_width"]
+            * self.params.hyperion_params.ispyb_params.microns_per_pixel_y
+            / 1000,
+        )
```

Comment:
Should: Is the only reason we need the grid scan info for constructing the comment? It seems messy to make this object here, use it once, then use the same parameters to make a `DataCollectionGridInfo` straight after. Could we change `construct_comment_for_gridscan` to take a `DataCollectionGridInfo` or to just take the parameters it needs then we can get rid of `GridScanInfo`?

Reply:
I will get rid of it